K8s prereq timeout

rata opened the pull request:
Having a different timeout for prerequisites is useful for us, as some jobs we run take more than what we want to wait for a deploy.

For backwards compatibility, the default value is the same as KUBE_WAIT_FOR_LIVE (so no workflow is affected unless explicitly changed).

@grosser What do you think? This is useful for us, and should not change anything for people not using it. And the code is not very complicated.

/cc @zendesk/samson

### Tasks
 - [ ] :+1: from team

### References
 - Jira link:

### Risks
- Level: Low

rata committed:
Use another ENV variable for k8s prereq timeout

For backwards compatibility, the default value is the same as
KUBE_WAIT_FOR_LIVE. But it can be adjusted to whatever is needed.

rata committed:
Add missing documentation for KUBE_WAIT_FOR_LIVE

rata commented on plugins/kubernetes/app/models/kubernetes/deploy_executor.rb:
```diff
@@ -72,7 +73,7 @@ def execute(*)
     # check all pods and see if they are running
     # once they are running check if they are stable (for apps only, since jobs are finished and will not change)
-    def wait_for_resources_to_complete(release_docs)
+    def wait_for_resources_to_complete(release_docs, timeout = WAIT_FOR_LIVE)
```
Maybe the default value can be removed here, just noticed 😱

grosser replied:
prefer keyword args and no default value

grosser commented:
looks good
would a project setting be better so it is not globally configured ?

grosser commented:
I guess global default is enough for now ... project setting would mean a
new column / form-field and can be an additional PR if needed

On Fri, Oct 12, 2018 at 8:38 AM rata <notifications@github.com> wrote:

> On Fri, Oct 12, 2018 at 08:00:50AM -0700, Michael Grosser wrote:
> > looks good
>
> Thanks!
>
> > would a project setting be better so it is not globally configured ?
>
> Oh, like a global default and a per project setting? That would be great,
> yes.
> How would that be? Do you have anything in mind? :)
>
> >
> >
> > --
> > You are receiving this because you authored the thread.
> > Reply to this email directly or view it on GitHub:
> > https://github.com/zendesk/samson/pull/2996#issuecomment-429355924
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/zendesk/samson/pull/2996#issuecomment-429368364>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AAAsZ825D3oHOqZCvKV_zZb1cCNg1DsUks5ukLdvgaJpZM4XZmqW>
> .
>

rata commented:
@grosser cool, thanks!
@adammw thanks!

Removed the default arg and fixed tests. Now all test pass, except `TASK=fossa`. Is it unrelated to this PR? The log didn't point anything related to this, AFAICS.

grosser commented on plugins/kubernetes/app/models/kubernetes/deploy_executor.rb:
```diff
@@ -6,6 +6,7 @@
 module Kubernetes
   class DeployExecutor
     WAIT_FOR_LIVE = ENV.fetch('KUBE_WAIT_FOR_LIVE', 10).to_i.minutes
+    WAIT_FOR_PREREQ = ENV.fetch('KUBE_WAIT_FOR_PREREQ', 10).to_i.minutes
```
should fallback to WAIT_FOR_LIVE when not set ?
`WAIT_FOR_PREREQ = ENV.fetch('KUBE_WAIT_FOR_PREREQ', WAIT_FOR_LIVE / 1.minute).to_i.minutes`

rata replied:
Yeah, makes sense. The code gets a little tricker with the division and conversion, IMHO. Wonder if can be made simpler...

Would this work:
```
WAIT_FOR_LIVE = ENV.fetch('KUBE_WAIT_FOR_LIVE', 10)
WAIT_FOR_PREREQ = ENV['KUBE_WAIT_FOR_PREREQ'] ||  WAIT_FOR_LIVE
WAIT_FOR_LIVE = WAIT_FOR_LIVE.to_i.minutes
WAIT_FOR_PREREQ = WAIT_FOR_PREREQ.to_i.minutes
```

But let me know which one you prefer (or just modify it, if GitHub default lets you change my branch) or if you can think of something simpler :)


Thanks!
